cedar-dafny/validation/types.dfy

```diff
@@ -69,7 +69,24 @@ module validation.types {
   }
 
   datatype AttrType = AttrType(ty: Type, isRequired: bool)
-  type RecordType = map<Attr,AttrType>
+  datatype OpenTag = OpenAttributes | ClosedAttributes
+  datatype RecordType = RecordType(
+    attrs: map<Attr,AttrType>,
+    // Indicates whether a value having this record type may have attributes
+    // beyond those lists in `attrs` (open), or if it must match `attrs`
+    // exactly (closed). In principal, any record type, including an entity
+    // attributes record type, may be closed or open, but the type of all record
+    // literals and any record type written in the schema is always closed. A
+    // least upper bound between record types will tend to be open, but it may
+    // be closed if the constituent record types are closed and have exactly the
+    // same attributes with a least upper bound existing between corresponding
+    // attributes.
+    openAttributes: OpenTag
```

Comment:
Nit: this field name is weird to me because `openAttributes` could be set to `OpenAttributes` or `ClosedAttributes`. Given that the current name of the type is `OpenTag`, I think the field name `tag` would make more sense.